src/openzeppelin/tests/test_erc721.cairo

```diff
@@ -0,0 +1,475 @@
+    assert(ERC721::_owners::read(TOKEN_ID()) == ZERO(), 'Low level ownership');
+
+    ERC721::_mint(RECIPIENT(), TOKEN_ID());
+
+    assert(ERC721::_exists(TOKEN_ID()), 'Token should exist');
+    assert(ERC721::_owners::read(TOKEN_ID()) == RECIPIENT(), 'Low level ownership');
+
+    ERC721::_burn(TOKEN_ID());
+
+    assert(!ERC721::_exists(TOKEN_ID()), 'Token should not exist');
+    assert(ERC721::_owners::read(TOKEN_ID()) == ZERO(), 'Low level ownership');
+}
+
+///
+/// approve & _approve
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_approve_from_owner() {
+    setup();
+
+    set_caller_address(OWNER());
+    ERC721::approve(SPENDER(), TOKEN_ID());
+    assert(ERC721::get_approved(TOKEN_ID()) == SPENDER(), 'Spender not approved correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_approve_from_operator() {
+    setup();
+
+    set_caller_address(OWNER());
+    ERC721::set_approval_for_all(OPERATOR(), true);
+
+    set_caller_address(OPERATOR());
+    ERC721::approve(SPENDER(), TOKEN_ID());
+    assert(ERC721::get_approved(TOKEN_ID()) == SPENDER(), 'Spender not approved correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected = ('ERC721: unauthorized caller', ))]
+fn test_approve_from_unauthorized() {
+    setup();
+
+    set_caller_address(OTHER());
+    ERC721::approve(SPENDER(), TOKEN_ID());
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected = ('ERC721: approval to owner', ))]
+fn test_approve_to_owner() {
+    setup();
+
+    set_caller_address(OWNER());
+    ERC721::approve(OWNER(), TOKEN_ID());
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected = ('ERC721: invalid token ID', ))]
+fn test_approve_nonexistent() {
+    ERC721::approve(SPENDER(), TOKEN_ID());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test__approve() {
+    setup();
+
+    ERC721::_approve(SPENDER(), TOKEN_ID());
+    assert(ERC721::get_approved(TOKEN_ID()) == SPENDER(), 'Spender not approved correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected = ('ERC721: approval to owner', ))]
+fn test__approve_to_owner() {
+    setup();
+
+    ERC721::_approve(OWNER(), TOKEN_ID());
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected = ('ERC721: invalid token ID', ))]
+fn test__approve_nonexistent() {
+    ERC721::_approve(SPENDER(), TOKEN_ID());
+}
+
+///
+/// set_approval_for_all & _set_approval_for_all
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_set_approval_for_all_true() {
+    set_caller_address(OWNER());
+    ERC721::set_approval_for_all(OPERATOR(), true);
+    assert(ERC721::is_approved_for_all(OWNER(), OPERATOR()), 'Operator not approved correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_set_approval_for_all_false() {
+    set_caller_address(OWNER());
+    ERC721::set_approval_for_all(OPERATOR(), false);
+    assert(!ERC721::is_approved_for_all(OWNER(), OPERATOR()), 'Operator not approved correctly');
```

Comment:
given `false` is the default value for approvals, we should probably set it to `true` to properly test this function's behavior